src/agent/block_store_services/block_store_azure.js

```diff
@@ -192,38 +180,36 @@ class BlockStoreAzure extends BlockStoreBase {
             count: 0
         };
         let failed_to_delete_block_ids = [];
-        return P.map_with_concurrency(10, block_ids, block_id => {
+        dbg.log0('block_store_azure._delete_blocks block_ids: ', block_ids);
+
+        return P.map_with_concurrency(10, block_ids, async block_id => {
                 const block_key = this._block_key(block_id);
                 let info;
-                return P.fromCallback(callback =>
-                        this.blob.getBlobProperties(
-                            this.container_name,
-                            block_key,
-                            callback)
-                    )
-                    .then(info_arg => {
-                        info = info_arg;
-                    })
-                    .then(() => P.fromCallback(callback =>
-                        this.blob.deleteBlob(
-                            this.container_name,
-                            block_key,
-                            callback)
-                    ))
-                    .then(() => {
-                        const data_size = Number(info.contentLength);
-                        const noobaablockmd = info.metadata.noobaablockmd || info.metadata.noobaa_block_md;
-                        const md_size = (noobaablockmd && noobaablockmd.length) || 0;
-                        deleted_storage.size -= (data_size + md_size);
-                        deleted_storage.count -= 1;
-                    })
-                    .catch(err => {
-                        if (err.code !== 'BlobNotFound') {
-                            failed_to_delete_block_ids.push(block_id);
-                        }
-                        dbg.warn('BlockStoreAzure _delete_blocks failed for block',
-                            this.container_name, block_key, err);
-                    });
+                try {
+                    const blob_client = azure_storage.get_blob_client(this.container_client, block_key);
+                    const info_arg = await blob_client.getProperties();
+                    dbg.log1('block_store_azure._delete_blocks info_arg: ', info_arg, 'block_key: ', block_key);
+
+                    info = info_arg;
```

Comment:
```suggestion
                   const  info = info_arg;
```

Reply:
I just delete the info and used info_arg instead